.github/workflows/release.yml

```diff
@@ -0,0 +1,59 @@
+# Copyright 2024 The Sigstore Authors
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#      http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+name: Release
+
+on:
+  tags:
+    - v*
```

Comment:
there's also a `release` trigger which you can further restrict by `published` type if that interest you more than `tags`. Which is what the example workflow at the end of https://docs.pypi.org/trusted-publishers/using-a-publisher/ does.

Reply:
Oh, that's a good idea. I sent #302 since `tags` should be under `push`.

Now we have a decision to make: do we want to push a tag and automatically create the release, or create the release and then trigger the workflow for the other steps? I have used either in other repos and can't decide right now

Reply:
I guess, when in doubt we should follow `sigstore-python`: https://github.com/sigstore/sigstore-python/blob/b5b9e3159c7eca5dd153201fad2c060b09b4d924/.github/workflows/release.yml#L3-L6

Will update #302

Reply:
I've used the tag approach in Go just due to the VCS based immutable nature of modules. But following `sigstore-python` SGTM.

Another aspect to consider is access control. You can protect tags with [tag protection rules](https://docs.github.com/en/repositories/managing-your-repositorys-settings-and-features/managing-repository-settings/configuring-tag-protection-rules). 

I'm not sure there is similar protection for releases?

Reply:
I think access control might require us to switch to tags? Let's test this on the next release (I'll ping you)